Otus-DevOps-2020-11/alexeyhead_infra
# # Completed HW No. 4

- [x] Basic HW
 - [x] Task with *

## In the process done:
  - Create new branch cloud-testapp
  - Install and setup yc CLI
  - Create host with yc CLI
  - On VM: update apt, install ruby, bundler and MongoDB
  - Deploy, run and check application
  - Performed items 3-5 using scripts install_ruby.sh, install_mongodb.sh and deploy.sh
  - Performed items 3-5 using startup script install_reddit_app_vm.sh

## How to run a project:
  - Run install_reddit_app_vm.sh
  or using CLI:

  ```
  yc compute instance create \
  --name reddit-app \
  --hostname reddit-app \
  --memory=4 \
  --create-boot-disk image-folder-id=standard-images,image-family=ubuntu-1604-lts,size=10GB \
  --network-interface subnet-name=default-ru-central1-a,nat-ip-version=ipv4 \
  --metadata serial-port-enable=1 \
  --ssh-key ~/.ssh/appuser.pub
  --metadata-from-file user-data=./metadata.yml
  ```

## How to check if it works:
  - Follow link: http://testapp_IP:testapp_port

## PR checklist
  - [x] Put a label with the homework number
  - [x] Put a topic with a homework theme
